src/exanb/compute/include/exanb/compute/compute_cell_particles.h

```diff
@@ -169,19 +166,14 @@ namespace exanb
     using CellsAccessorT = std::conditional_t< has_external_or_optional_fields , std::remove_cv_t<std::remove_reference_t<decltype(grid.cells_accessor())> > , CellsPointerT >;
     using PForFuncT = ComputeCellParticlesFunctor<CellsAccessorT,FuncT,FieldTupleT,std::make_index_sequence<sizeof...(FieldAccT)> >;
     
-    size_t N;
-    
-    if(number_filled_cells == 0)
-    {
-    	const IJK dims = grid.dimension();
```

Comment:
define before this scope so that dims is known for `PForFuncT pfor_func = { cells , dims , gl , func , cpfields, filled_cells };`.